main.js

```diff
@@ -392,6 +392,51 @@ if(typeof System !== "undefined") {
 	addTilde(System);
 }
 
+function addContextual(loader){
+  loader._contextualModules = {};
+
+  loader.setContextual = function(moduleName, definer){
+    // `definer` could be a function or could be a moduleName
+		if (typeof definer === 'string') {
+			this._contextualModules[moduleName] = this['import'](definer);
```

Comment:
Like we talked about yesterday, doing the import here means that the import happens when `setContextual` is called, even if no one is going to be using this module. We should just save the string name here and import it in `normalize` when it's needed.